pkg/daemon/daemon.go

```diff
@@ -255,6 +259,25 @@ func (d *Daemon) sharedFusedevMount(rafs *Rafs) error {
 			d.ConfigFile(rafs.SnapshotID))
 	}
 
+	if config.IsKeyringEnabled() {
+		image, err := registry.ParseImage(rafs.ImageID)
```

Comment:
It's would be great if we can wrap this code snippet into a helper function which tries to fill the auth into config string

Reply:
Do you mean to wrap the code for parsing images and get the `keyChain` to a function?

Reply:
> Do you mean to wrap the code for parsing images and get the `keyChain` to a function?

Done.